src/x86_64/ftrace.s

```diff
@@ -86,16 +93,63 @@ extern __current_ftrace_graph_return
 
 global ftrace_stub
 global mcount
+global return_to_handler
+
 mcount:
-    cmp qword [__current_ftrace_trace_fn], ftrace_stub
-    jnz trace
+    cmp qword [__ftrace_function_fn], ftrace_stub
```

Comment:
Could these just be function vectors that are called unconditionally?

Reply:
Yeah we could do that. The idea here though is that doing three compares is quicker than doing function calls that just immediately ret, which will issue a few extra instructions to manipulate the stack and frame pointers